pkg/util/helper/binding.go

```diff
@@ -125,21 +121,29 @@ func (a *Dispenser) TakeByWeight(w ClusterWeightInfoList) {
 }
 
 // GetStaticWeightInfoListByTargetClusters constructs a weight list by target cluster slice.
-func GetStaticWeightInfoListByTargetClusters(tcs []workv1alpha2.TargetCluster) ClusterWeightInfoList {
-	weightList := make(ClusterWeightInfoList, 0, len(tcs))
-	for _, result := range tcs {
+func GetStaticWeightInfoListByTargetClusters(available, scheduled []workv1alpha2.TargetCluster) ClusterWeightInfoList {
+	weightList := make(ClusterWeightInfoList, 0, len(available))
+	for _, targetCluster := range available {
+		var lastReplicas int32
+		for _, scheduledCluster := range scheduled {
+			if targetCluster.Name == scheduledCluster.Name {
+				lastReplicas = scheduledCluster.Replicas
+				break
+			}
+		}
 		weightList = append(weightList, ClusterWeightInfo{
-			ClusterName: result.Name,
-			Weight:      int64(result.Replicas),
+			ClusterName:  targetCluster.Name,
+			Weight:       int64(targetCluster.Replicas),
+			LastReplicas: lastReplicas,
 		})
 	}
 	return weightList
 }
 
 // SpreadReplicasByTargetClusters divides replicas by the weight of a target cluster list.
-func SpreadReplicasByTargetClusters(numReplicas int32, tcs, init []workv1alpha2.TargetCluster) []workv1alpha2.TargetCluster {
```

Comment:
tcs and init is confusing, so I just renamed it

Reply:
I don't think so, `tcs` means `target clusters`, it's not the same meaning of `available`.

Reply:
done